src/controllers/company.number.controller.ts

```diff
@@ -52,6 +57,39 @@ const route = async (req: Request, res: Response, next: NextFunction): Promise<v
       templateName: templatePaths.COMPANY_NUMBER,
     });
   }
+
+  const companyNumber: string = req.body.companyNumber;
+  try {
+    logger.info(`Retrieving company profile for company number ${companyNumber}`);
+    const token: string = req.chSession.accessToken() as string;
+
+    // TODO: Place this call somewhere more global so that it not called on check company as well
+    const company: PTFCompanyProfile = await getCompanyProfile(companyNumber, token);
+
+    await sessionService.createPromiseToFileSession(req.chSession, company.companyNumber);
```

Comment:
add a check so that you only create a session if one doesn't exist

Reply:
Added the check to the create function itself.

Reply:
The check added was causing problems so removing this.

Reply:
As we are not doing this in extensions-web and so will need investigating so I suggest we do this on another ticket.

Reply:
Added update function which can be called from the company number controller, removed check from create function.

Reply:
can you move the create session to a middleware function to be called from app.ts after the authorisation? that way all routes will be covered and not just this one.